_15thnight/models.py

```diff
@@ -246,16 +250,42 @@ def to_provider_json(self, provider):
         return extend(self.to_json(), dict(
             responses=Response.get_by_user_and_alert(provider, self),
             needs=[
-                need.to_provider_json(provider) for need in self.needs \
-                    if need.service.id in service_ids
+                need.to_provider_json(provider) for need in self.needs
+                if need.service.id in service_ids
             ]
         ))
 
+    def is_open(self):
+        """Check if all needs have been met."""
+        for need in self.needs:
+            if not need.resolved:
+                return False
+        return True
+
+    def get_gender(self):
+        """
+        Return a formatted string with the gender if male/female.
+        Blank string if unspecified.
+        """
+        gender = '' if self.gender == 'unspecified' else self.gender
+        g_string = " %s" % gender
+        return g_string
+
+    def url(self):
```

Comment:
In `core.py` you're referencing this as a property. Let's make `url` a property function.